lib/EnsEMBL/REST/Model/Eqtl.pm

```diff
@@ -0,0 +1,152 @@
+
+extends 'Catalyst::Model';
+with 'Catalyst::Component::InstancePerContext';
+
+# Overwrite potentially, check Catalyst doc
+# Assign all these to $self->context
+sub build_per_context_instance {
+  my ($self, $c, @args) = @_;
+  return $self->new({ registry => $c->model('Registry'), %$self, @args });
+
+}
+
+=head2 fetch_qetl
+
+  Arg [1]    : HASHREF $constraints
+  Example    : $eqtls = $eqtl_adaptor->fetch_eqtl($constraints);
+  Description: Validates tissue constraint.
+               Fetches all statistics for from the DB using the constraints supplied.
+  Returntype : HASHREF
+  Exceptions : Throws if constraint is not present in Database
+  Caller     : general
+  Status     : Stable
+
+=cut
+
+sub fetch_eqtl {
+  my ($self, $constraints ) = @_;
+
+  # fails in registry if species is wrong/not available in Ensembl
+  my $user_species    = $constraints->{species};
+  my $ensembl_species = $self->{'registry'}->get_alias($user_species);
+
+  if(! $ensembl_species) {
+    Catalyst::Exception->throw("Species '$user_species' not recognised. Check your URL");
+  }
+
+  my $eqtl_a = $self->{'registry'}->get_eqtl_adaptor($ensembl_species);
+
+  $self->_validate_stable_id ($eqtl_a, $constraints->{stable_id});
+  $self->_validate_tissuei   ($eqtl_a, $constraints->{tissue});
+  $self->_validate_statistic ($eqtl_a, $constraints->{statistic});
+
+  my $results = $eqtl_a->fetch( {
+        gene          => $constraints->{stable_id},
+        tissue        => $constraints->{tissue},
+        snp           => $constraints->{variant_name},
+        statistic     => $constraints->{statistic},
+      });
+
+  return $results;
+}
+
+=head2 _validate_stable_id
+
+  Arg [1]    : Bio::EnsEMBL::HDF5::EQTLAdaptor
+  Arg [2]    : $stable_id
+  Example    : $self->_validate_stable_id($eqtl_a, $constraints->{stable_id});
+  Description: Validates if stable_id exists in Database.
+  Returntype : None
+  Exceptions : Throws if stable_id is not present in Database
+  Caller     : general
+  Status     : Stable
+
+=cut
+
+sub _validate_stable_id {
+  my ($self, $eqtl_a, $stable_id) = @_;
+  # Check for EnsemblIDs with more/less than 11 digits
+  if (defined $stable_id){
+    if( ($stable_id =~ /ENS[A-Z]+[0-9]{11}/)  and (! exists $eqtl_a->{gene_ids}->{$stable_id}) ) {
+      Catalyst::Exception->throw("Stable ID '$stable_id' not present in the current Ensembl release");
+    }
+  }
+}
+
+=head2 _validate_tissue
+
+  Arg [1]    : Bio::EnsEMBL::HDF5::EQTLAdaptor
+  Arg [2]    : $tissue
+  Example    : $self->_validate_tissue($eqtl_a, $constraints->{tissue});
+  Description: Validates if tissue exists in Database.
+  Returntype : None
+  Exceptions : Throws if tissue is not present in Database
+  Caller     : general
+  Status     : Stable
+
+=cut
+
+sub _validate_tissue {
+  my ($self, $eqtl_a, $tissue) = @_;
+  if (defined $tissue){
+    if(! exists $eqtl_a->{tissue_ids}->{$tissue}) {
+      my $tissues = join(", ", map { $_ } sort keys %{$eqtl_a->{tissue_ids}});
```

Comment:
You can push this into a Moose accessor. Make it lazy and set the default to this code